axelrod/tests/strategies/test_axelrod_second.py

```diff
@@ -694,3 +695,162 @@ def test_strategy(self):
         actions += [(C, C), (C, D), (D, C)]
         actions += [(C, D)] # Raise flag
         actions += [(C, C)] # Use flag to change outcome
+        self.versus_test(Time_Passer, expected_actions=actions)
+
+
+class TestHarrington(TestPlayer):
+    name = "Harrington"
+    player = axelrod.Harrington
+    expected_classifier = {
+        'memory_depth': float('inf'),
+        'stochastic': True,
+        'makes_use_of': set(),
+        'long_run_time': False,
+        'inspects_source': False,
+        'manipulates_source': False,
+        'manipulates_state': False
+    }
+
+    def test_strategy(self):
```

Comment:
Thanks for moving the chi-squared attribute. This test function is pretty long now, you might consider breaking it up into several test functions. (optional)

Reply:
You could also target specific functions with their own tests, e.g. for `detect_streak` and friends.